src/I18n/DateFormatTrait.php

```diff
@@ -66,29 +73,35 @@ protected function _formatObject(
         }
 
         $timezone = $date->getTimezone()->getName();
-        if ($timezone === '+00:00' || $timezone === 'Z') {
-            $timezone = 'UTC';
-        } elseif ($timezone[0] === '+' || $timezone[0] === '-') {
-            $timezone = 'GMT' . $timezone;
-        }
-
-        $formatter = datefmt_create(
-            $locale,
-            $dateFormat,
-            $timeFormat,
-            $timezone,
-            $calendar,
-            $pattern
-        );
-        if (!$formatter) {
-            $key = "{$locale}.{$dateFormat}.{$timeFormat}.{$timezone}.{$calendar}.{$pattern}";
-            throw new CakeException(
-                'Your version of icu does not support creating a date formatter for ' .
-                "`$key`. You should try to upgrade libicu and the intl extension."
+        $key = "{$locale}.{$dateFormat}.{$timeFormat}.{$timezone}.{$calendar}.{$pattern}";
+
+        if (!isset(static::$formatters[$key])) {
+            if ($timezone === '+00:00' || $timezone === 'Z') {
+                $timezone = 'UTC';
+            } elseif ($timezone[0] === '+' || $timezone[0] === '-') {
+                $timezone = 'GMT' . $timezone;
+            }
+
```

Comment:
Normalize `$timezone` before generating `$key`.
```suggestion
        if ($timezone === '+00:00' || $timezone === 'Z') {
            $timezone = 'UTC';
        } elseif ($timezone[0] === '+' || $timezone[0] === '-') {
            $timezone = 'GMT' . $timezone;
        }

        $key = "{$locale}.{$dateFormat}.{$timeFormat}.{$timezone}.{$calendar}.{$pattern}";

        if (!isset(static::$formatters[$key])) {
```

Reply:
We could do that. I thought that originally it was possibly a micro optimization, as it is a little faster with thousands of calls, talking like `0.004` seconds faster with 10.000 calls.